tests/test_association.py

```diff
@@ -0,0 +1,357 @@
+                        dataId=dataId_dict)
+
+        (self.tmp_file, self.db_file) = tempfile.mkstemp(
+            dir=os.path.dirname(__file__))
+        assoc_db_config = AssociationDBSqliteConfig()
+        assoc_db_config.db_name = self.db_file
+        assoc_db = AssociationDBSqliteTask(config=assoc_db_config)
+        assoc_db.create_tables()
+
+        dia_collection = DIAObjectCollection(create_test_dia_objects(5))
+        assoc_db.store(dia_collection, True)
+        assoc_db.close()
+
+    def tearDown(self):
+        del self.assoc_task
+
+        if os.path.exists(self.testDir):
+            shutil.rmtree(self.testDir)
+        if hasattr(self, "butler"):
+            del self.butler
+
+        del self.tmp_file
+        os.remove(self.db_file)
+
+    def test_valid_from_metadata(self):
+        """Verify that association information can be recovered from metadata.
+        """
+        # Insert data into the task metadata.
+        test_assoc_result = pipeBase.Struct(
+            n_updated_dia_objects=5,
+            n_new_dia_objects=6,
+            n_unassociated_dia_objects=7,)
+        self.assoc_task._add_association_meta_data(test_assoc_result)
+        metadata = self.assoc_task.getFullMetadata()
+
+        meas = measure_number_new_dia_objects(
+            metadata,
+            "association",
+            "association.numNewDIAObjects")
+        self.assertIsInstance(meas, Measurement)
+        self.assertEqual(
+            meas.metric_name,
+            lsst.verify.Name(metric="association.numNewDIAObjects"))
+        self.assertEqual(meas.quantity, 6 * u.count)
+
+        meas = measure_number_unassociated_dia_objects(
+            metadata,
+            'association',
+            'association.fracUpdatedDIAObjects')
+        self.assertIsInstance(meas, Measurement)
+        self.assertEqual(
+            meas.metric_name,
+            lsst.verify.Name(
+                metric='association.fracUpdatedDIAObjects'))
+        self.assertEqual(meas.quantity, 7 * u.count)
+
+        meas = measure_fraction_updated_dia_objects(
+            metadata,
+            'association',
+            'association.fracUpdatedDIAObjects')
+        self.assertIsInstance(meas, Measurement)
+        self.assertEqual(
+            meas.metric_name,
+            lsst.verify.Name(metric='association.fracUpdatedDIAObjects'))
+        self.assertEqual(meas.quantity, 5 / (5 + 7) * u.dimensionless_unscaled)
+
+
+    def test_valid_from_butler(self):
+        """ Test the association measurements that require a butler.
+        """
+        meas = measure_number_sci_sources(
+            self.butler,
+            dataId_dict=dataId_dict,
+            metric_name='ip_diffim.numSciSrc')
+        self.assertIsInstance(meas, Measurement)
+        self.assertEqual(
+            meas.metric_name,
+            lsst.verify.Name(metric='ip_diffim.numSciSrc'))
+        self.assertEqual(meas.quantity, 10 * u.count)
```

Comment:
Where does the 10 come from?

Reply:
Changed this to a variable stored in self.